src/sentry/ingest/transaction_clusterer/tasks.py

```diff
@@ -62,8 +62,11 @@ def cluster_projects(projects: Sequence[Project]) -> None:
         if features.has("organizations:transaction-name-clusterer", project.organization):
             with sentry_sdk.start_span(op="txcluster_project") as span:
                 span.set_data("project_id", project.id)
+                tx_names = list(redis.get_transaction_names(project))
+                if len(tx_names) < redis.MAX_SET_SIZE:
+                    return
                 clusterer = TreeClusterer(merge_threshold=MERGE_THRESHOLD)
-                clusterer.add_input(redis.get_transaction_names(project))
+                clusterer.add_input(iter(tx_names))
```

Comment:
Wouldn't the type checker accept this as well? A `List` is an `Iterable` AFAIK.

```suggestion
                clusterer.add_input(tx_names)
```

Reply:
Heh, you're right. I was thinking of `Iterator` instead of `Iterable`. Addressed in [fefe8c3](https://github.com/getsentry/sentry/pull/46503/commits/fefe8c3e50b23f4ebeb196f33a7beb186068f61d).